### DISCUSSION TITLE
API Call syntax

### DISCUSSION THREAD

**Discussion by natrlhy** [teslamate-org/teslamate#3079] [Q&A] [answered]
I'm trying to learn a bit more with API formatting and using curl. How would I properly format this API call (after getting an access token successfully:

https://tesla-api.timdorr.com/vehicle/commands/sentrymode

`curl --request POST --header 'Authorization: Bearer longtokenstringhere' 'https://owner-api.teslamotors.com/api/1/vehicles/1492656798636023/command/set_sentry_mode'`

**Comment by cwanja** [comment, marked as answer]
Totally Googling here. Not in front of a terminal to try it myself. 

But it looks like you need to use the `-d` flag and pass the parameter and desired value. 

Turn on:
```
{
  "on": "true"
}
```

Turn off:
```
{
  "on": "false"
}
```

All together, changing false to true and visa versa: 
`curl --request POST --header 'Authorization: Bearer longtokenstringhere' -d '{ "on": "false"}' 'https://owner-api.teslamotors.com/api/1/vehicles/id/command/set_sentry_mode'`

**Reply by cwanja** [reply]
Sources 
https://rapidapi.com/blog/api-glossary/post/

https://www.tutorialspoint.com/unix_commands/curl.htm

**Reply by natrlhy** [reply]
I tried sending the data payload, too:

`{"response":null,"error":"not_a_JSON_request","error_description":""}`

**Reply by cwanja** [reply]
Can you try adding `-H 'Content-Type: application/json'` to the command?

**Reply by cwanja** [reply]
Just got time in-front of a terminal. Got this work. Need to be specific and particular about the single and double quotes you used. Some fonts will not work:
```
curl "https://owner-api.teslamotors.com/api/1/vehicles/{id}/command/set_sentry_mode" -H "Authorization: Bearer {your token ID}" -X “POST” -H "Content-Type: application/json" -d '{ "on": "false"}'
```

Edit: fixed it further with quotes. Here I was able to honk the horn:
```
curl "https://owner-api.teslamotors.com/api/1/vehicles/{id}/command/honk_horn" -H "Authorization: Bearer {your token}" -X "POST"

{"response":{"result":true,"reason":""}}
```

**Reply by natrlhy** [reply]
Thanks! I tinkered with the quotes a bit but confirmed this works! Much appreciated and saved notes, too!
